### ISSUE TITLE
Should static ServiceProvider methods be final?

### ISSUE THREAD

**Issue by keilw** [unitsofmeasurement/unit-api#126]
All static accessor methods in `ServiceProvider` like `available()`, `current()` or `of()` are `public` and `static`, but not `final`.  Shouldn't it be `final` as well, so these methods cannot be overriden by subclasses?

**Comment by desruisseaux** [comment]
Static method can not be overridden. It is possible to declare a method of the same name in subclasses, but it is more similar to overloading than overriding: `ServiceProvider.current()` (for instance) we still invoke the `ServiceProvider` implementations, regardless subclasses. Adding a `final` keyword indeed prevent subclasses from defining a method of the same name and signature, but that method would have been visible only to users who write explicitly `TheSubClass.current()` for example.

So in summary: no objection on my side, but I'm not sure it would provide real benefit neither.

**Comment by keilw** [comment]
Then it probably doesn't hurt. Should have no impact on e.g. the JAR size either.
